CONTRIBUTING.md

```diff
@@ -0,0 +1,39 @@
+# Contributing to Pulumiverse
+
+:+1::tada: Thanks for taking the time to contribute! :tada::+1:
+
+The following is a set of guidelines for contributing. These are just
+guidelines, not rules, so use your best judgment and feel free to propose
+changes to this document in a pull request.
+
+## Issues and Pull Requests
+
+- If you're not sure about adding something, open an issue to discuss it.
+- Feel free to open a Pull Request early so that a discussion can be had as
+  changes are developed.
+
+## Commit Messages and Pull Request Titles
+
+We use the
+[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
```

Comment:
I have never used this approach myself. To be honest, I find this a burden and would make me turn away from contributing.

Reply:
I've only recently come across this convention, and I can see a lot of benefits to it. I can see how it would be helpful, and how hooks can validate this convention to quickly get us in the habit.
However there's nothing in the link that offers recommendations on how to deal with probably the most common merge technique in open source projects: PR from a fork with many commits which do not follow this convention.
If the convention was adopted, would this need to be addressed too? I supposed a squash merge with a new commit message would make it work...

Reply:
Seems like this might not be an issue, because of the way the bot is implemented. Its documented goal is
> to ensure that every PR has just enough semantic information to be able to trigger a release when appropriate. The goal is to gather this semantic information in a way that doesn't make life harder for project contributors

Reply:
@cobraz I find this a burden because it is something I never remember when committing code. Why would we want to push this onto contributors? What is the value of conventional commits? A generated changelog file is not enough for me to opt-in on this convention.

Reply:
@tenwit I agree, we should probably address that too. In most cases, I squash and merge forks with many commits. In those cases, the pull request title could be the only thing be _conventional_. I've used this method for about three-four years now, and have most of my CI/CD systems hooked up.

@ringods You don't have to remember when you're committing; as mentioned above. For the most part, this is something that we need to ensure when merging pull requests. We also don't need every commit to be conventional commits compliant, just those who contain fixes, new features, and breaking changes.

I don't have any strong opinions here on whether we should have this or not. However, IMHO, there is more to this than generated changelogs. I can spin up a few arguments here:

- Improved visibility; makes it easier for people to read through commit logs and understand what things are. By pointing to a specification, I find that people tend to think more about how the commits look like and the general wording. E.g. Instead of `fix dep circle`, I generally see more of this kinds `fix: remove manhattan function causing dep cicle`.
- CI/CD is much easier. In many of my repositories, I use Google's [Release Please](https://github.com/googleapis/release-please) or [Semantic Release](https://github.com/semantic-release/semantic-release). These are releasing tools that use conventional commits to prepare or release software. To give you an example, [have a look at this last release at generate-runtypes](https://github.com/cobraz/generate-runtypes/pull/126). This is an automated pull request that Release Please created, which when merged triggered a release to NPM (no humans interacted with NPM, bumping of versions or figuring out which version to bump to).

Lastly, I just wanna point out that most of this burden will be put on active contributors (people who review and merge pull requests).

Reply:
> it is something I never remember when committing code
For the initial adoption of someone else's repo to Pulumiverse, we have to take what we get. There's no way around this. I don't think this is a problem: the main purpose of Conventional Commits in an open source project is to remove the burden of creating changelogs; and since this is the initial adoption of the project, there are (from Pulumiverse's point of view) no changes to log.
For a Pulumiverse repo that still being primarily maintained by its original developer(s), then we can add to the GitHub conversation to request that they adopt the convention, add the enforcing actions, etc. Time and leniency are important - we _could_ update the docs to be explicit about this, but I don't think we need to.
For a repo that has no primary maintainer (that is, Pulumiverse itself is the hands-on maintainer), then we should have git hooks / Husky  to reject commits, with useful docs on how to format them correctly. This is how I learned about Conventional Commits, and I find that it doesn't add a significant overhead.

In my opinion, the value of automatically-generated changelogs outweighs the burden of having to remember to write those changelogs at time of making the change. No one wants to have to trawl through old commits during each release in order to manually generate changelogs. And hopefully, no one wants to maintain an open source project without changelogs.

Reply:
I think this is great. I’ve been using CC for years and as a consumer and contributor of a lot of OSS, it’s very useful when I find other projects that use it too. It makes checking out the recent changes much easier and means the project is very likely to have a decent Changelog.

Reply:
@tenwit, I agree on keeping what the project already has and eventually _migrate_. Should we add something about this? 

I don't think we need to. 🤷‍♂️